File: cCarRentalSystem/Client/HttpClients/ClientInterfaces/IReservationService.cs
using Domain.DTOs;
using Domain.Models;

namespace HttpClients.ClientInterfaces;

public interface IReservationService
{
    Task<Reservation> CreateAsync(ReservationCreationDTO reservationToCreate);
    Task ReturnReservationAsync(int reservationId); 
    Task CancelReservationAsync(int reservationId);
    Task<IEnumerable<Reservation>> GetReservationsAsync();
    Task<Reservation> GetReservationById(int reservationId);
    Task<IEnumerable<Reservation>> GetReservationsByCarId(int carId);
    Task<IEnumerable<Reservation>> GetReservationsByUsername(string username);
    Task<bool> IsCarAvailable(int carId, DateTime startDate, DateTime endDate);
    
}

File: CarRentalService/src/main/java/com/example/carrentalservice/service/Reservation/IReservationService.java
package com.example.carrentalservice.service.Reservation;

import com.example.carrentalservice.model.Reservation;

import java.util.List;

public interface IReservationService
{
    Reservation createReservation(Reservation reservation);

    Reservation returnReservation(int id);

    List<Reservation> getReservations();
    Reservation getReservationById(int id);
    List<Reservation> getReservationsByCarId(int carId);
    List<Reservation> getReservationsByUserName(String userName);




}
